packages/data-context/src/DataContext.ts

```diff
@@ -349,8 +349,11 @@ export class DataContext {
 
   onError = (err: Error) => {
     if (this.isRunMode) {
-      // console.error(err)
-      throw err
+      if (this.lifecycleManager?.runModeExitEarly) {
```

Comment:
This is a bit of a hack, when we implementing run mode for unified (next week) we can probably start to make this control flow less convoluted.